src/main/java/com/puppycrawl/tools/checkstyle/api/AbstractCheck.java

```diff
@@ -294,6 +294,15 @@ public final String getLine(int index) {
         return context.get().fileContents.getLine(index);
     }
 
+    /**
+     * Returns the name of the class.
+     *
+     * @return The class name.
+     */
+    public String getClassName() {
+        return getClass().getName();
```

Comment:
do we need special method for this ?

Reply:
> Comparator.comparing(AbstractCheck::getClassName)

I do not know how to do this without a method and need help with determining another way.
`Object::getClass::getClassName` does not work.

Reply:
@strkkk , can you help us here ?

Reply:
`Comparator.comparing(check -> check.getClass().getName())`?

Reply:
Comparable is much harder to add as this was causing issues with pitest and `hashCode` is a bit more complex to control ordering. I also wasn't sure if this would be consider a drastic change for this issue and if @romani would be against it.

Reply:
@strkkk Apparently that does not compile successfully.
````
        return new TreeSet<>(Comparator.comparing(check -> check.getClass().getName())
                .thenComparing(AbstractCheck::getId)
                .thenComparing(AbstractCheck::hashCode));
````

`The type AbstractCheck does not define getId(Object, Object) that is applicable here`.

Reply:
@rnveach it will compile in this way (a bit uglier but no need for extra method)
```
final Comparator<AbstractCheck> nameComparing = Comparator.comparing(check -> check.getClass().getName());
        new TreeSet<>(nameComparing
            .thenComparing(AbstractCheck::getId)
            .thenComparing(AbstractCheck::hashCode));
```

Reply:
This works. Just waiting on https://github.com/checkstyle/checkstyle/pull/9254#discussion_r574936883

Reply:
I am ok with this.

No new inheritances for AbstractCheck in this PR :) .

Reply:
Done.